api/server/router/build/build_routes.go

```diff
@@ -161,7 +162,23 @@ func parseVersion(s string) (types.BuilderVersion, error) {
 }
 
 func (br *buildRouter) postPrune(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
-	report, err := br.backend.PruneCache(ctx)
+
+	kd, err := strconv.Atoi(r.FormValue("keep-duration"))
+	if err != nil {
+		return errors.Wrap(err, "keep-duration expects an integer number in seconds")
+	}
+	ks, err := strconv.ParseFloat(r.FormValue("keep-storage"), 32)
+	if err != nil {
+		return errors.Wrap(err, "keep-storage expects a floating point number in MB")
+	}
+
+	opts := types.BuildCachePruneOptions{
+		All:          httputils.BoolValue(r, "all"),
```

Comment:
For my clarification:
Isnt `all` mutually exclusive with the Keep* options? 
Also, what happens when `KeepDuration` is met but `KeepStorage` isnt or vice versa?